core/arch/arm/kernel/kern.ld.S

```diff
@@ -233,6 +228,12 @@ SECTIONS
 		*(.data .data.* .gnu.linkonce.d.*)
 		. = ALIGN(8);
 		KEEP(*(SORT(.scattered_array*)));
+#ifdef CFG_DT
+		. = ALIGN(8);
```

Comment:
Same problem here as with scattered arrays. While looking at this I wonder why the dtdrv stuff isn't implemented using scattered arrays.